### PR TITLE
#67 error handling infinite loop issue 

### PR DESCRIPTION
**Pull request by ishaileshmishra**
error handling infinite loop issues & code quality improved

### CODE REVIEW

File: src/main/java/com/contentstack/sdk/ContentstackPlugin.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.contentstack.sdk;
+
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+
+public interface ContentstackPlugin {
+
+    default void onRequest(Stack stack, Request request) {
```
Comment (uttamukkoji) [inline src/main/java/com/contentstack/sdk/ContentstackPlugin.java:9]: @ishaileshmishra onRequest should return Request object

Comment (ishaileshmishra) [reply src/main/java/com/contentstack/sdk/ContentstackPlugin.java:9]: Noted @uttamukkoji

File: src/test/java/com/contentstack/sdk/TestLivePreview.java [hunk 2]
```diff
@@ -73,7 +74,7 @@ void testStackEnableLivePreviewQuery() throws Exception {
         HashMap<String, String> hashMap = new HashMap<>();
         hashMap.put("live_preview", "hash167673");
         hashMap.put("content_type_uid", "contentType");
-        stack.livePreviewQuery(hashMap);
+        //stack.livePreviewQuery(hashMap);
```
Comment (uttamukkoji) [inline src/test/java/com/contentstack/sdk/TestLivePreview.java:77]: @ishaileshmishra remove commented code

File: src/test/java/com/contentstack/sdk/TestLivePreview.java [hunk 3]
```diff
@@ -87,14 +88,14 @@ void testStackEnableLivePreviewEntry() throws Exception {
...
```
Comment (uttamukkoji) [inline src/test/java/com/contentstack/sdk/TestLivePreview.java:91]: @ishaileshmishra Remove commented code

**Review by uttamukkoji — APPROVED** [review]

**Review by aravindbuilt — APPROVED** [review]

**Pull request merged**